app/models/role.rb

```diff
@@ -1,10 +1,16 @@
 class Role < ActiveRecord::Base
   belongs_to :resource, polymorphic: true
   has_and_belongs_to_many :users, join_table: :users_roles
-
+  before_destroy :cancel
```

Comment:
And there is no way to tell rolify not to do that other then making it impossible in the model?